src/components/EventCard/FavoriteButton.js

```diff
@@ -0,0 +1,45 @@
+import React, { Component, Fragment } from 'react';
+import { inject, observer } from 'mobx-react';
+import { withRouter } from 'react-router';
+import {
+  withStyles,
+} from '@material-ui/core';
+import { FavoriteBorder, Favorite } from '@material-ui/icons';
+
+import styles from './styles';
+
+@withStyles(styles, { withTheme: true })
+@withRouter
+@inject('store')
+@observer
+class FavoriteButton extends Component {
+  state = {
+    forceRefreshFlag: false,
```

Comment:
why do you need this? we should be able to handle refreshes based on the observing changes in the store.

Reply:
Yes let's avoid hacky solutions to make things work. Should be able to get this to work using Mobx and observable values.